### PR TITLE
Refactor the single_end meta thing to be pulled from the ENA metadata.

### PR DESCRIPTION
**Pull request by mberacochea**
Include the Fetch tool metadata into the multiqc report

### CODE REVIEW

File: modules/local/fetchtool_reads.nf [hunk 1]
```diff
@@ -10,22 +10,29 @@ process FETCHTOOL_READS {
...
-    -v $private_study $args
+    -p ${study_accession} \\
+    -ru ${reads_accession} \\
+    -c ${fetchtool_config} \\
+    -v ${private_study} ${args}
+
+    library_strategy=\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 7)
+    library_layout=\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 5)
+
+    cp download_folder/${study_accession}/${study_accession}.txt download_folder/${study_accession}/fetch_tool_mqc.tsv
```
Comment (KateSakharova) [inline modules/local/fetchtool_reads.nf:35]: mv? :)

**Review by KateSakharova — COMMENTED** [review]
I think taking metadata from ENA is good. but we should probbaly add an ability to user/prefect to push correct library_strategy or library_layout to pipeline. Waiting for ENA to fix data can take time (and we had cases wherethey must contact users but users are unknown/do not reply..). 
Basically, first iteration of pipeline should be with ENA metadata. If something failed, we checked - we should be able to specify fixed args and push assembly and do not delay it. 
In my view, we also need to add `library_strategy` or `library_layout` to params

File: modules/local/fetchtool_reads.nf (hunk 1 above)
Comment (mberacochea) [reply modules/local/fetchtool_reads.nf:35]: I will remove the `_mqc` file, I couldn't make MultiQC to accept it withut that prefix

**Comment by mberacochea** [comment]
> I think taking metadata from ENA is good. But we should probably add an ability to user/prefect to push correct library_strategy or library_layout to pipeline. Waiting for ENA to fix data can take time (and we had cases where they must contact users but users are unknown/do not reply..). Basically, first iteration of pipeline should be with ENA metadata. If something failed, we checked - we should be able to specify fixed args and push assembly and do not delay it. In my view, we also need to add `library_strategy` or `library_layout` to params

Indeed.. I wil add 2 more parameters for that (same logic, if they are there then it will ignore the metadata)

File: modules/nf-core/fastp/main.nf [hunk 2]
```diff
@@ -29,7 +29,8 @@ process FASTP {
     script:
     def single_end = reads.collect().size() == 1
     def args = task.ext.args ?: ''
-    def polyA = trim_polyA ? "--trim_poly_x" : ''
+    // Our team addition to handle metaT
```
Comment (chrisAta) [inline modules/nf-core/fastp/main.nf:32]: Maybe add a patch diff file for this too like you have for seqkit?

File: nf-test.config [hunk 3]
```diff
@@ -3,5 +3,5 @@ config {
     testsDir "tests"
     workDir ".nf-test"
     configFile "tests/nextflow.config"
-    profile "test,docker"
+    profile "docker"
```
Comment (chrisAta) [inline nf-test.config:6]: should the docker profile be preferred over singularity?

File: tests/main.nf.test [hunk 4]
```diff
@@ -3,20 +3,119 @@ nextflow_pipeline {
...
+                assert success
+                assert trace.succeeded().count{ task -> task.name.contains("SPADES") } == 0
+                assert trace.succeeded().count{ task -> task.name.contains("MEGAHIT") } == 1
+                assert trace.tasks().size() == 17
+            }
+        }
+
+    }
+
+    // test("metaSPAdes - single end") {
```
Comment (chrisAta) [inline tests/main.nf.test:54]: remove commented test

**Review by chrisAta — CHANGES_REQUESTED** [review]
Looks good to me thank you!! Just a few minor comments

**Review by Ales-ibt — APPROVED** [review]
It looks Ok to me, thank you.

File: modules/nf-core/fastp/main.nf (hunk 2 above)
Comment (mberacochea) [reply modules/nf-core/fastp/main.nf:32]: yes.. I was having some issues with nf-core using the schema plugin

File: nf-test.config (hunk 3 above)
Comment (mberacochea) [reply nf-test.config:6]: it simplifies the execution in GA, no need to setup singularity... that is the only reason I left it like that

File: tests/main.nf.test (hunk 4 above)
Comment (mberacochea) [reply tests/main.nf.test]: I've fixed those tests now :)

File: modules/nf-core/fastp/main.nf
Comment (Ge94) [inline modules/nf-core/fastp/main.nf:1]: Remove "Add --interleaved_in in config." comment (line 38) - this was done

**Review by Ge94 — COMMENTED** [review]
Nothing to say really, huge work. Thank you!

There is just one comment that could be addressed

**Review by Ge94 — APPROVED** [review]

**Review by chrisAta — APPROVED** [review]

**Comment by mberacochea** [comment]
Perfect. I have enough :+1: to merge... :shipit:

**Pull request merged**